src/room/InCallView.tsx

```diff
@@ -328,6 +354,76 @@ export const InCallView: FC<InCallViewProps> = ({
   const audioOutputSwitcher = useBehavior(vm.audioOutputSwitcher$);
   useSubscription(vm.autoLeave$, onLeave);
 
+  // When we enter timeout or decline we will leave the call.
+  useEffect((): void | (() => void) => {
+    if (callPickupState === "timeout") {
+      onLeave();
+    }
+    if (callPickupState === "decline") {
+      void pickupPhaseAudio
+        ?.playSound("decline")
+        .then(() => {
```

Comment:
Presumably, also this needs moving after catch. 
```suggestion
        .finally(() => {
```

Otherwise a failed sound will prevent a leave.